compute_sdk/globus_compute_sdk/sdk/client.py

```diff
@@ -99,18 +109,21 @@ def __init__(
 
         self._task_status_table: dict[str, dict] = {}
 
-        # if a login manager was passed, no login flow is triggered
-        if login_manager is not None:
-            self.login_manager: LoginManagerProtocol = login_manager
-        # but if login handling is implicit (as when no login manager is passed)
-        # then ensure that the user is logged in
+        if app and login_manager:
+            raise ValueError(
+                "'app' and 'login_manager' are mutually exclusive; please choose one."
+            )
+        elif login_manager and not app:
+            self.login_manager = login_manager
```

Comment:
`and not app` is redundant, given the first branch.

Reply:
Ah, good catch.